spring-grpc-spring-boot-autoconfigure/src/test/java/org/springframework/grpc/autoconfigure/client/GrpcClientObservationAutoConfigurationTest.java

```diff
@@ -0,0 +1,88 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.grpc.autoconfigure.client;
+
+import io.micrometer.core.instrument.binder.grpc.ObservationGrpcClientInterceptor;
+import io.micrometer.observation.ObservationRegistry;
+import org.assertj.core.api.Condition;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.FilteredClassLoader;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.grpc.client.GlobalClientInterceptor;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for the {@link GrpcClientObservationAutoConfiguration}.
+ */
+class GrpcClientObservationAutoConfigurationTests {
+
+	private final ApplicationContextRunner baseContextRunner = new ApplicationContextRunner()
+		.withConfiguration(AutoConfigurations.of(GrpcClientObservationAutoConfiguration.class));
+
+	private ApplicationContextRunner validContextRunner() {
+		return new ApplicationContextRunner()
+			.withConfiguration(AutoConfigurations.of(GrpcClientObservationAutoConfiguration.class))
+			.withBean("observationRegistry", ObservationRegistry.class, Mockito::mock);
+	}
+
+	@Test
+	void whenObservationRegistryNotOnClasspathAutoConfigSkipped() {
+		this.validContextRunner()
+			.withClassLoader(new FilteredClassLoader(ObservationRegistry.class))
+			.run((context) -> assertThat(context).doesNotHaveBean(GrpcClientObservationAutoConfiguration.class));
+	}
+
+	@Test
+	void whenObservationGrpcClientInterceptorNotOnClasspathAutoConfigSkipped() {
+		this.validContextRunner()
+			.withClassLoader(new FilteredClassLoader(ObservationGrpcClientInterceptor.class))
+			.run((context) -> assertThat(context).doesNotHaveBean(GrpcClientObservationAutoConfiguration.class));
+	}
+
+	@Test
+	void whenObservationRegistryNotProvidedThenAutoConfigSkipped() {
+		this.baseContextRunner
+			.run(context -> assertThat(context).doesNotHaveBean(GrpcClientObservationAutoConfiguration.class));
+	}
+
+	@Test
+	void whenObservationPropertyEnabledThenAutoConfigNotSkipped() {
+		this.validContextRunner()
+			.withPropertyValues("spring.grpc.client.observation.enabled=true")
+			.run(context -> assertThat(context).hasSingleBean(GrpcClientObservationAutoConfiguration.class));
+	}
+
+	@Test
+	void whenObservationPropertyDisabledThenAutoConfigIsSkipped() {
+		this.validContextRunner()
+			.withPropertyValues("spring.grpc.client.observation.enabled=false")
+			.run(context -> assertThat(context).doesNotHaveBean(GrpcClientObservationAutoConfiguration.class));
+	}
+
+	@Test
+	void whenAllConditionsAreMetThenInterceptorConfiguredAsExpected() {
```

Comment:
I like this test you have added. It checks if the interceptor bean is registered and if its marked as global interceptor. I think the test may be a bit easier to follow by using `hasEntrySatisifying` - something like:
```java
@Test
void whenAllConditionsAreMetThenInterceptorConfiguredAsExpected() {
this.contextRunner().run((context) -> {
	assertThat(context).hasSingleBean(ObservationGrpcClientInterceptor.class);
	assertThat(context.getBeansWithAnnotation(GlobalClientInterceptor.class))
			.hasEntrySatisfying("observationGrpcClientInterceptor", 
					(bean) -> bean.getClass().isAssignableFrom(ObservationGrpcClientInterceptor.class));
});
}
```
Wdyt?

Reply:
Yes I agree. 

I copied the tests from [GrpcServerObservationAutoConfigurationTests](https://github.com/spring-projects-experimental/spring-grpc/blob/729940d62c15fb198e5e3232fec7f06031f80ede/spring-grpc-spring-boot-autoconfigure/src/test/java/org/springframework/grpc/autoconfigure/server/GrpcServerObservationAutoConfigurationTests.java).
 Do you want me to adjust the test for the server too? Should that be done in a separate PR?

Reply:
Thats ok @NotBjoggisAtAll , I will update that one separately.